src/config.ts

```diff
@@ -313,3 +307,70 @@ export const SOFTWARE_VERSION = (() => {
     // into <pre> or <code> where it will create an unnecessary newline.
     return /^(.*)$/m.exec(versionFile)?.at(0) ?? defaultText;
 })();
+
+
+// Command line related functions
+
+/**
+ * Grabs an option from the command line and checks if it exists.
+ * @param arg Option name
+ * @returns True if the option is present, otherwise false.
+ */
+function isCommandLineOptionPresent(args: string[], arg: string): boolean {
+    return args.includes(arg);
+}
+
+/**
+ * Grabs an option's value from program options if it exists, otherwise returns undefined.
+ * @param arg Option name
+ * @param throwOnInvalid If true or undefined, throws an error if the option is present but has no value. If false, returns undefined.
+ * @returns The value passed to the option, or undefined if the option is not specified.
+ * @throws Error if the option is present but has no value.
+ */
+function getCommandLineStringOption(args: string[], arg: string, throwOnInvalid?: boolean): string | undefined {
+    // We don't want to throw if the option is not present
+    if(!isCommandLineOptionPresent(args, arg)) {
+        return undefined;
+    }
+
+    const optionIndex = args.indexOf(arg);
+    if (optionIndex === -1) {
+        return undefined;
```

Comment:
Both of these checks are surely redundant, we only need the second one right?

Reply:
I would remove the second one, personally (consistency with the path function, and i'd say its fair to assume optionIndex would never be -1 if it's present, right?)

Reply:
Well, it does make me feel uneasy, since we are using `optionIndex`. If the implementation for `isCommandLineOptionPresent` breaks, then our now naive use of `optionIndex` will be confusing to whoever has to discover it is now `-1`.